react-basic-practice/src/components/CartBar/index.jsx

```diff
@@ -2,57 +2,83 @@ import React from 'react';
 import Button from '../common/Button';
 import CartItem from '../CartItem';
 import closeButton from '@/assets/images/iconButton/btn-close.png';
+import { useCart } from '@/contexts/CartProvider';
+import { useToast } from '@/contexts/ToastProvider';
 
 import './index.css';
+import Toast from '../Toast';
 
-const data = [
-  {
-    idProduct: 2,
-    quantity: 3,
-    name: 'hal earrings',
-    price: 25,
-  },
-  {
-    idProduct: 1,
-    quantity: 4,
-    name: 'lira earrings',
-    price: 20,
-  },
-];
-
-const CartBar = () => {
-  // const {
-  //   onClose, data,
-  // } = props;
-  const onDeleteCartItem = () => {};
-  const onHandleCloseNav = () => {
-    // onClose(false);
+const CartBar = (props) => {
+  const { handleCloseCart } = props;
+  const { listItem, setListItem, updateItemCart } = useCart();
+  const { toast, setToast } = useToast();
+
+  const onDeleteCartItem = (e) => {
+    try {
+      const idSelected = e.target.closest('.cart-item').id;
+      const updateCarts = listItem.listProducts.filter(
+        (cartItem) => cartItem.idProduct !== Number(idSelected),
+      );
+
+      const cartUser = {
+        id: 1,
+        listProducts: updateCarts,
+      };
+
+      setListItem(cartUser);
+      updateItemCart(cartUser);
+
+      setToast({
+        openPopup: true,
+        status: 'success',
+        message: 'The item removed from your shopping bag',
+      });
+    } catch {
+      setToast({
+        openPopup: true,
+        status: 'error',
+        message: 'Remove from cart failed, please try again',
+      });
+    }
+  };
+
+  const handleClose = () => {
+    setToast({ ...toast, openPopup: false });
   };
 
   return (
     <aside className="cart-bar">
-      <div className="overlay" />
       <div className="main">
         <div className="header">
           <h2 className="title">Shopping bag</h2>
           <Button
-            onClick={onHandleCloseNav}
+            onClick={handleCloseCart}
             type="button"
             className="btn btn-close"
             icon={closeButton}
           />
         </div>
         <div className="list-cart-item">
           {
-          data.map((item) => (
-            <CartItem
-              item={item}
-              handleDeleteCartItem={onDeleteCartItem}
-            />
-          ))
+            listItem.listProducts.map((item) => (
+              <CartItem
```

Comment:
What prop shold you have when looping?

Reply:
That's the key, I'll use the id for the key